tactical-microgrid-standard/common/HeartbeatDataReaderListenerImpl.h

```diff
@@ -0,0 +1,60 @@
+#ifndef HEARTBEAT_DATAREADER_LISTENER_IMPL
+#define HEARTBEAT_DATAREADER_LISTENER_IMPL
+
+#include <dds/DCPS/LocalObject.h>
+#include <dds/DdsDcpsSubscriptionC.h>
+
+#include <functional>
+
+class HeartbeatDataReaderListenerImpl
+  : public virtual OpenDDS::DCPS::LocalObject<DDS::DataReaderListener> {
+public:
+  HeartbeatDataReaderListenerImpl(std::function<void(const tms::Heartbeat&, const DDS::SampleInfo&)> cb = nullptr)
+    : callback_(cb)
+  {}
+
+  virtual ~HeartbeatDataReaderListenerImpl() {}
+
+  virtual void on_requested_deadline_missed(DDS::DataReader_ptr,
+                                            const DDS::RequestedDeadlineMissedStatus&)
+  {
+    ACE_DEBUG((LM_INFO, "(%P|%t) INFO: HeartbeatDataReaderListenerImpl::on_requested_deadline_missed\n"));
+  }
```

Comment:
This is just a suggestion, but if there's going to be more listeners where we're not going to use most of the methods, maybe use a common base sorta like the one I started to work on in https://github.com/OpenDDS/OpenDDS/pull/3460.

Reply:
I agree. I think there'll be more listeners when more topics are added.